src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,119 @@
+import java.sql.SQLException;
+import java.sql.ResultSet;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private final static int FIRST_PARAMETER = 1;
+    private final static int SECOND_PARAMETER = 2;
+    private final static int THIRD_PARAMETER = 3;
+    private final static int AFFECTED_ROW = 1;
+
+    @Override
+    public Book create(Book book) {
+        String sql = "INSERT INTO books (title, price) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+            PreparedStatement statement = connection
+                    .prepareStatement(sql, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(FIRST_PARAMETER, book.getTitle());
+            statement.setBigDecimal(SECOND_PARAMETER, book.getPrice());
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < AFFECTED_ROW) {
+                throw new RuntimeException("Expected to insert at least one row, "
+                        + "but inserted zero rows ");
+            }
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add new book to DB " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setLong(FIRST_PARAMETER, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                Book book = new Book();
+                book.setId(id);
+                book.setTitle(resultSet.getString("title"));
+                book.setPrice(resultSet.getBigDecimal("price"));
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get book by index " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+        String sql = "SELECT * FROM books";
+        try (Connection connection = ConnectionUtil.getConnection();
+        PreparedStatement statement = connection.prepareStatement(sql)) {
+            ResultSet resultSet = statement.executeQuery(sql);
+            while (resultSet.next()) {
+                Book book = new Book();
+                book.setId(resultSet.getLong("id"));
+                book.setTitle(resultSet.getString("title"));
+                book.setPrice(resultSet.getBigDecimal("price"));
```

Comment:
i think it will be more clearly, if you create extra helper method that will parse your resultSet data to Book, because your logic is repeating 2 times, in findAll() and findById() Methods.